main.py

```
@@ -43,12 +43,26 @@ def apply_for_single_session(
         By.XPATH, "//input[@type='text' and @placeholder='シリアルナンバー']"
     )
 
-    print("Filling code")
+    logprint("Filling code")
     serial_code_box.send_keys(code)
 
     apply_button = driver.find_element(By.XPATH, "//button[text()='お申込みへ']")
     apply_button.click()
 
+    # Attempt of an exception handling method aimed to react when the code is either wrong or used.
+    time.sleep(3)
+    error_messages = driver.find_elements(By.CLASS_NAME, "varidation")
+    # print(len(error_messages))
+    if error_messages:
+        error_msg = error_messages[1].text
+        if error_msg == "利用回数を超えたためお申込みできません。":
+            logprint(f"Code {code} has been used for {session_name} before. Please check again.")
+            return 2
+        elif error_msg == "申し込み情報が正しくありません。":
+            logprint(f"Code {code} is incorrect. Please check again.")
+            return 1
```

I'm not saying to change it now
but the ``time.sleep(3)`` irks me
there's probably some way to do it with selenium in a much cleaner fashion

mostly cause it might take more than 3 seconds to the page to load which could cause the error checking to fail
and then maybe throw an exception that would make the code stop running

so a selenium-based solution might be better
but that's for later

---

Tackled this with the new commit